packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -266,6 +267,77 @@ module.exports = ({ strapi }) => ({
     return mappedEntity;
   },
 
+  async publishMany(opts, uid) {
+    const params = {
+      ...opts,
+      data: {
+        [PUBLISHED_AT_ATTRIBUTE]: new Date(),
+      },
+    };
+
+    const query = transformParamsToQuery(uid, params);
+    const entitiesToUpdate = await strapi.db.query(uid).findMany(query);
+    // No entities to update, return early
+    if (!entitiesToUpdate.length) {
+      return null;
+    }
+
+    // Validate entities before publishing, throw if invalid
+    await Promise.all(
+      entitiesToUpdate.map((entityToUpdate) =>
+        strapi.entityValidator.validateEntityCreation(
+          strapi.getModel(uid),
+          entityToUpdate,
+          {
+            isDraft: true,
+          },
+          entityToUpdate
+        )
+      )
+    );
+    // Everything is valid, publish
+    const publishedEntitiesCount = await strapi.db
+      .query(uid)
+      .updateMany({ ...query, data: params.data });
+
+    // Get the updated entities since updateMany only returns the count
+    const publishedEntities = await strapi.db.query(uid).findMany(query);
```

Comment:
This query should contain this populate 😃 :
```js
    const populatedQuery = {
      ...query,
      populate: populateRelations
        ? getDeepPopulate(uid, {})
        : getDeepPopulate(uid, { countMany: true, countOne: true }),
    };
```

When publishing an entity (or mutating one), we normally send populated relations to webhooks.

Reply:
Got it thanks! I'll update

Reply:
@Marc-Roig concerning this suggestion: https://github.com/strapi/strapi/pull/16595#discussion_r1183497814 I'll still need the query again here to refetch the updated entries, so I guess it would mean passing the entities (after the permission check) + the query.

Reply:
Woldn't it be enough having the entities ID ? 
Maybe I am missing a use case!, but if the user doesn't have the permission what should be the behaviour? Not publish the entities without permission to do so, or throw an error and not publish any?

Reply:
Yes! I recently discovered this was possible too
![image](https://user-images.githubusercontent.com/20578351/235942027-30b2aef7-eb25-41d9-a89a-bf36df3fe945.png)
You can define custom conditions on every action, so a given user might be able to publish some entities and not publish other ones.

Reply:
ohhh, thanks I didn't know that!